ReleaseNotes/ThunderReleaseNotes_R5.3.md

```diff
@@ -0,0 +1,408 @@
+Starting 5.3 they will be included by default when empty (as [] and {}) (this on request of the Comcast Thunder user base).
+
+When it is desired they are not included when empty they can be made optional (with OptionalType< >(var)) in the interface, then when not set they will not be included.
+
+Note this is not considered breaking as both are valid JSON (and it is on specific request).
+
+### Feature: Allow inclusion of enum and POD from other IDL header file
+
+It is now possible to use an enum or POD (struct with data members) defined in one IDL header file in another by including the other header file with @insert
+
+### Feature: Enable enum <-> string conversion for non JSON-RPC interfaces
+
+Previously the enum to string (and vice versa) conversion tables were only generated when the enum was used in an interface used in JSON-RPC generation (so with @json tag).
+Starting 5.3 they can be generated for COM-RPC only interfaces as well. 
+For this use the @encode:text tag
+
+NOTE: self still add to Thunder documentation
+
+### Feature: support optional iterators:
+
+Iterators in the IDL header file can now also be of OptionalType<>.
+
+### Feature: support restrict for iterators
+
+The @restrict tag can now, next to strings, arrays and std::vector, also be used to set the minimum and maximum allowed size for iterator types in the IDL header file.
+
+### Feature: allow only lower bound restrict for strings
+
+When an input string in a method in the IDL is not allowed to be empty (but it is not desirable to set a maximum length, if that is the case the @restrict:x..y tag can be used) it can be flagged with the @restrict:nonempty tag.
+If the string is empty this will already generate an error when validating the input in the generated proxy stub code.
+
+NOTE: self still add to Thunder documentation 
+
+### Feature: collated iterators
+
+Iterators in the past always got their values one by one via the COM-RPC interface leading to overhead in an IPC situation as multiple COM-RPC calls are needed while most of the times the iterators were used to in the end get all the values.
+It is now possible to make all the iterators get all the values in one go and after that the COM-RPC calls to get the values will only be local calls.
+To enable this mode for all iterators you can pass the flag --collated-iterators to the proxy stub generator.
+With this mode enabled one can by specifying the tag @ASINTERFACE with an iterator make that one work like it did in the past and get the values one by one.
+
+Note in Thunder 6 this mode might become the default, that has not been done yet to give the opportunity to put the @ASINTERFACE at the iterators that potentially deal with huge amounts of data where this might pose a too big of a memory penalty.
+
+NOTE: is @interface with capitals? I guess put it as after the iterator name? also if you put it with the typedef (you would not need to repeat it)
```

Comment:
no, all tags are lowercase

Reply:
Wait for the update, this was work in progress  :)